src/GraphQL/Resolvers/NameFieldResolver.cs

```diff
@@ -25,7 +25,7 @@ public class NameFieldResolver : IFieldResolver
         public static NameFieldResolver Instance { get; } = new NameFieldResolver();
 
         /// <inheritdoc/>
-        public object Resolve(IResolveFieldContext context) => Resolve(context?.Source, context?.FieldAst?.Name);
+        public object Resolve(IResolveFieldContext context) => Resolve(context.Source, context.FieldDefinition.Name);
```

Comment:
I think this change is sound, but it does break a bunch of tests where `FieldDefinition` isn't set.  I didn't spend time to figure it out for sure.